### DISCUSSION TITLE
Beams seen as slurs - what to change? and: multi-selection possible?

### DISCUSSION THREAD

**Discussion by hmmueller** [Audiveris/audiveris#522] [Q&A] [answered]
In the following segment (of a Bach arrangement), quite a few beams are overlooked:

![OverlookedBeams](https://user-images.githubusercontent.com/1670656/143681737-080e06a2-afda-46cd-89eb-9e3045d5c8e7.JPG)

a) Which options might help Audiveris to "see" these beams (I looked into the options with search term "beam", but was overwhelmed by the number and short explanations of all matching parameters, I confess)?

b) Is it possible to multi-select all these "glyphs" and assign "Beam" to them?

Thank you!

(BTW: I found Audiveris only a few days ago - it has helped me to make a MXL score from a symphony by [O.C.E.v.Kospoth](https://de.wikipedia.org/wiki/Otto_Carl_Erdmann_von_Kospoth) in a few hours, instead of days ... this is a great program!!)

H.M.

**Comment by hbitteur** [comment]
This is generally the symptom of a sheet with too few beams (which might not be the case according to your small excerpt).

Anyway, check the log window for the message about beams in SCALE step. There you should find information about beam mean thickness. If the value is prefixed with "Extra" (for extrapolated), then this is the culprit.

Could you please post the first lines of your logged messages (the ones related to the SCALE step), so that we can check this.

**Reply by hmmueller** [reply]
Thx for quick reply! 
The only messages with "scale" and "beam" in them are (one for each of the 5 pages):

[IMSLP272748#1] Scale{line(1,1,2) interline(20,21,21) beam(11)}

No "extra" to be seen :( ... I have added the pdf, omr and log here (they are not that large):

[IMSLP272748.zip](https://github.com/Audiveris/audiveris/files/7611775/IMSLP272748.zip)

H.M.

**Comment by hmmueller** [comment]
Related to my initial question b): I tried to assign "BEAM" to the glyphs that were not assigned or wrongly assigned SLUR. However, this worked only for some of them - but not for others, which would not connect to both stems on the left and right. In the editor, one can see that there are gaps on the left and right of an assigned beam:

![BeamAssignmentTrouble](https://user-images.githubusercontent.com/1670656/143686506-cd73f4e0-d41c-448c-b111-a860b735640a.JPG)

And in a created MXL- shown here in MuseScore -, only the right note is a sixteenth:

![BeamAssignmentTroubleMXL](https://user-images.githubusercontent.com/1670656/143686539-15f0e2d2-c716-4037-bf5f-0f1751c23077.JPG)

How can I "connect" this beam on both sides?

H.M.

**Comment by hbitteur** [comment]
I'm debugging your case right now...

Early results: all your beams  are correctly recognized, hence my initial guess on beam scale was wrong.
The problem is that many beam hooks (if not all) don't get connected to stems.
I still don't know why precisely, score quality is OK.
It seems that each of these beam hooks is in exclusion with a full beam inferred from the same underlying glyph. 
I need to investigate further.
Give me some time.

Regarding your 2nd question: you cannot select several glyphs to assign each to a separate Inter with the same shape.
When you select several glyphs, the engine builds a new "compound glyph" on-the-fly, and then handles this compound as a single big glyph. Here, this is not what you expected.

General remark: you should post these topics on Audiveris [issues forum](https://github.com/Audiveris/audiveris/issues) (rather than this one meant for general release discussion). This would be more convenient to share the information with all the other users.

**Reply by hmmueller** [reply]
Thanks for investigation!
Re 2nd question - also thanks!
And I'll post "issues" in the "issues" forum from now on.

(Still, all of Audiveris results are very helpful - repairing the resulting scores in some target editor is still quicker than writing everything from scratch ...)

H.M.

**Comment by hmmueller** [comment, marked as answer]
My problem is satisfyingly - for this single score - answered [here](https://github.com/Audiveris/audiveris/issues/524) by hbitteur (link added to clean up the messages mess I created ...).

H.M.

**Comment by hbitteur** [comment]
Issue now fixed via commit a754cfe134ef4214f9687d13d5bec2b2dc398206 (available on development branch).

Don't forget to undo my suggested workaround  (maxHookWidth in BeamsBuilder unit). 
To do do, use `Tools | Options` menu and on the line dedicated to maxHookWidth constant, simply uncheck the 'modif' box.
The constant should go back from 1.7 to 2.0
